File: Fertilizer.cs
using System.Collections.Generic;

public class Fertilizer : SourcedItem
{
    public static Fertilizer None = new Fertilizer("None", 0, 0);

    public int Quality { get; }
    public float Speed { get; }

    public Fertilizer(string name, int quality, float speed) : base(name)
    {
        Quality = quality;
        Speed = speed;
    }

    public Fertilizer(string name, int quality, float speed, Dictionary<Source, int> priceFrom) : base(name, priceFrom)
    {
        Quality = quality;
        Speed = speed;
    }
}

File: src/Types/Fertilizer.fs
namespace StardewValleyStonks

open Types

type Fertilizer =
  { Name: string
    Selected: bool
    Quality: int
    Speed: float
    PriceFrom: Map<Name<Source>, Price> }
  member this.Toggle = { this with Selected = not this.Selected }

type FertilizerSort =
  | ByName
  | Selected
  | Quality
  | Speed
  | Price

//[<RequireQualifiedAccess>]
module Fertilizer =
  let name fertilizer = fertilizer.Name
  let selected fertilizer = fertilizer.Selected
  let quality fertilizer = fertilizer.Quality
  let speed fertilizer = fertilizer.Speed

  let create
    name
    quality
    speed
    prices =
    { Name = name
      Selected = true
      Quality = quality
      Speed = speed
      PriceFrom = priceListToMap prices }